### ISSUE TITLE
VariableList should implement TryFrom<Vec> instead of truncating

### ISSUE THREAD

**Issue by benluelo** [sigp/ssz_types#12]
I was quite surprised to see that [`VariableList` implements `From<Vec>`](https://github.com/sigp/ssz_types/blob/main/src/variable_list.rs#L124) - this should be a `TryFrom` implementation instead. As per the [docs on From](https://doc.rust-lang.org/stable/std/convert/trait.From.html):

> Note: This trait must not fail. The From trait is intended for perfect conversions. If the conversion can fail or is not perfect, use [TryFrom](https://doc.rust-lang.org/stable/std/convert/trait.TryFrom.html).

**Comment by michaelsproul** [comment]
I agree this would be a nice change to make. We can add it to `ssz_types` first in a new v0.6 release, then integrate it into Lighthouse at a later date.

Lighthouse is also moving towards `milhouse` types instead of the ones in this crate. Our `tree-states` branch already removes usage of `From<Vec<_>>` because `milhouse` doesn't support it: https://github.com/sigp/milhouse/blob/e30719aaf1e4b46e3ff1962f7370965e68fa548b/src/list.rs#L45-L47.

**Comment by michaelsproul** [comment]
Sorry, missed this for 0.6 and 0.7. Will try to get it in for 0.8

**Comment by michaelsproul** [comment]
Closed by:

- https://github.com/sigp/ssz_types/pull/52
